pkg/prometheus/statefulset.go

```diff
@@ -405,8 +405,14 @@ func makeStatefulSetSpec(
 	}
 
 	// TODO(simonpasquier): check that the Prometheus version supports the flag.
-	if p.Spec.Web != nil && p.Spec.Web.PageTitle != nil {
-		promArgs = append(promArgs, monitoringv1.Argument{Name: "web.page-title", Value: *p.Spec.Web.PageTitle})
+	if p.Spec.Web != nil {
+		if p.Spec.Web.PageTitle != nil {
+			promArgs = append(promArgs, monitoringv1.Argument{Name: "web.page-title", Value: *p.Spec.Web.PageTitle})
+		}
+
+		if p.Spec.Web.MaxConnections != nil {
+			promArgs = append(promArgs, monitoringv1.Argument{Name: "web.max-connections", Value: fmt.Sprintf("%d", *p.Spec.Web.MaxConnections)})
```

Comment:
Not sure if I should've added a check on the prometheus version. This flag was introduced in prometheus 1.5.0 [1]

[1] https://github.com/prometheus/prometheus/blob/main/CHANGELOG.md#150--2017-01-23
[2] https://github.com/prometheus/prometheus/pull/2240

Reply:
Having removed support for `1.x` .  we don't need to check that IIUC

Reply:
Thanks for letting me know!